File: Geometry/point_location_test.py
t = int(input())
for _ in range(t):
    x1, y1, x2, y2, x3, y3 = map(int, input().split())

    if x1 == x2:
        if x3 == x1:
            print("TOUCH")
        elif y1 < y2:
            if x3 < x1:
                print("LEFT")
            else:
                print("RIGHT")
        else:
            if x3 < x1:
                print("RIGHT")
            else:
                print("LEFT")
        continue

    slope = (y2 - y1) / (x2 - x1)
    intercept = y1 - slope * x1

    ans = slope * x3 + intercept

    ans = round(ans, 6)

    if y3 == ans:
        print("TOUCH")
    elif y3 < ans:
        if x1 < x2:
            print("RIGHT")
        else:
            print("LEFT")
    else:
        if x1 < x2:
            print("LEFT")
        else:
            print("RIGHT")
    
    

File: Geometry/point_location_test.cpp
#include <bits/stdc++.h>

#define int long long
#define need_for_speed ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);

using namespace std;

int32_t main() {
    need_for_speed
    
    int n, x1, y1, x2, y2, x3, y3; cin >> n;
    while (n--)
    {
        cin >> x1 >> y1 >> x2 >> y2 >> x3 >> y3;
        int area = (x2 - x1) * (y3 - y1) - (x3 - x1) * (y2 - y1);
        if (area == 0) cout << "TOUCH\n";
        else if (area < 0) cout << "RIGHT\n";
        else cout << "LEFT\n";
    }
    
    return 0;
}